### ISSUE TITLE
Transfer rules_scala to bazel-contrib

### ISSUE THREAD

**Issue by meteorcloudy** [bazel-contrib/rules_scala#1616]
**Description:**

This issue proposes transferring the rules_scala repository from the `bazelbuild` GitHub organization to the `bazel-contrib` GitHub organization, which is owned by the [Community for Bazel](https://github.com/bazel-contrib/.github/blob/master/Community%20for%20Bazel%20Technical%20Charter.pdf) project under the Linux Foundation.

**GitHub Discussion:** https://github.com/orgs/bazelbuild/discussions/2

**Impacts:**

* GitHub issues & PRs will be transferred to the new repository.
* Branch protection rules will need to be transferred to the new repository.
* Individual maintainers will be transferred to the new repository.
* GitHub teams will NOT be transferred to the new repository and will need to be re-created.
* Bazel CI configurations will need to be updated to point to the new repository.

Read more on [Transferring a repository](https://docs.github.com/en/repositories/creating-and-managing-repositories/transferring-a-repository) from GitHub documentation.

**Checklist:**

* [x] Obtain approval from the current project maintainers (@liucijus, @simuons).
* [x] Obtain approval from the bazel-contrib SIG (@alexeagle).
* [x] Initiate the transfer process on GitHub (@meteorcloudy).
* [x] Update Bazel CI settings and confirm presubmit & postsubmit work (@meteorcloudy).

**Comment by alexeagle** [comment]
Notes from SIG meeting: this transfer was unanimously approved by a thumbs-up vote.

**Comment by simuons** [comment]
Approve. Thanks.

**Comment by liucijus** [comment]
👍🏻

**Comment by meteorcloudy** [comment]
Thanks for the approvals, I'll initiate the transfer now.

**Comment by meteorcloudy** [comment]
Just noticed rules_scala doesn't support Bzlmod yet, is that going to be a blocker? @alexeagle 

See 9 from https://bazel-contrib.github.io/SIG-rules-authors/hosting-policy.html#adding-criteria

**Comment by alexeagle** [comment]
Yes, thank you - we ought to hold off on this one in that case.

Of course it's just as feasible to fix for bzlmod after the transfer, but I think the SIG would like to know who is committed to doing that work.

**Comment by meteorcloudy** [comment]
Agree, it's better to clarify that since we are disabling WORKSPACE in Bazel 8.

**Comment by meteorcloudy** [comment]
Essentially blocking on https://github.com/bazelbuild/rules_scala/pull/1595

@liucijus @simuons Any plan to review and accept that PR?

**Comment by simuons** [comment]
@meteorcloudy I don't think that PR is complete. I'll be able to look at this next week.

**Comment by mbland** [comment]
BTW, I've just posted my willingness to contribute to Bzlmodification in https://github.com/bazelbuild/rules_scala/issues/1482#issuecomment-2389248501, if folks are cool with that. As stated there, I've a (mostly) working branch already.

**Comment by simuons** [comment]
Hi, @mbland, if you could take this that would be awesome. I promised to look at it a week ago but couldn't. So any help here would be greatly appreciated.

**Comment by mbland** [comment]
@simuons My pleasure! I'm very close to having [the bzlmod branch in my fork](https://github.com/mbland/rules_scala/tree/bzlmod) passing `./test_all.sh`. I'll be carving off other independent pieces shortly.

A question regarding Bazel 6 support: I'm making sure my changes are still compatible with Bazel 6 and WORKSPACE, but I've also got them working with the latest Bazel 7. I don't have a strong opinion, but I'm curious which of the following options sound good to you:

- Keep .bazelversion and .bazelci at 6.5.0, test Bazel 7 locally while developing.
- Bump .bazelversion and .bazelci to (currently) 7.3.2, test Bazel 6 locally while developing.
- Bump .bazelversion to 6.5.0 and add extra CI jobs for 7.3.2 (or vice versa).

**Comment by mbland** [comment]
FYI, a brief status update: https://github.com/bazelbuild/rules_scala/issues/1482#issuecomment-2407784993

Basically, I'm very close to being done with my first draft, and anticipate sending more pull requests soon.

**Comment by simuons** [comment]
Hi, @mbland, answering "regarding Bazel 6 support: ". I think we should use third option. Keep bazel 6 until bzlmod is supported. Once bzlmod is ready we will make bazel 6 release and switch to bazel 7. Does this sound good to you?

**Comment by mbland** [comment]
@simuons Sounds good to me!

**Comment by meteorcloudy** [comment]
Thanks to @mbland https://github.com/bazelbuild/rules_scala/issues/1482 is close to the finish line. I think bazel-contrib can accept the transfer now and it will make the publish to bcr process easier (a fixed release isn't necessary anymore).

@alexeagle @simuons @liucijus Please confirm if you are fine with the transfer.

**Comment by meteorcloudy** [comment]
Gently ping~  @simuons @liucijus

**Comment by meteorcloudy** [comment]
@simuons @liucijus Since this has been previously approved, I'll start with the transfer process tomorrow so that publishing rules_scala to BCR is easier.

**Comment by meteorcloudy** [comment]
I'm starting the transfer process now.

**Comment by meteorcloudy** [comment]
This is done, let me know if there is any issues.
